File: Leetcode/122-Best-Time-to-Buy-and-Sell-Stock-II/BestTimeToBuyAndSellStockII.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        profit = 0
        if len(prices)<2:
            return profit
        cost = prices[0]
        cur = 1
        while cur<len(prices):
            if prices[cur] < prices[cur-1]:
                profit = profit + prices[cur-1] - cost
                cost = prices[cur]
            cur+=1
        profit = profit + prices[cur-1] - cost
        return profit
    

File: Leetcode/122-Best-Time-to-Buy-and-Sell-Stock-II/BestTimeToBuyAndSellStockII.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var profit = 0
    if (!prices || prices.length<2) {
        return profit
    }
    var cost = prices[0]
    var cur = 1
    while (cur<prices.length) {
        if (prices[cur] < prices[cur-1]) {
            profit = profit + prices[cur-1] - cost
            cost = prices[cur]
        }
        cur++
    }
    profit = profit + prices[cur-1] - cost
    return profit
};
